BaseTools/Plugin/HostBasedUnitTestRunner/HostBasedUnitTestRunner.py

```diff
@@ -112,4 +112,123 @@ def do_post_build(self, thebuilder):
                                             "  %s - %s" % (case.attrib['name'], result.text))
                                         failure_count += 1
 
+            if thebuilder.env.GetValue("TOOL_CHAIN_TAG") == "GCC5":
+                self.gen_code_coverage_gcc(thebuilder)
+            elif thebuilder.env.GetValue("TOOL_CHAIN_TAG") == "VS2019":
+                self.gen_code_coverage_msvc(thebuilder)
+            else:
+                logging.info("Skipping code coverage. Only supported on GCC.")
+
         return failure_count
+
+    def gen_code_coverage_gcc(self, thebuilder):
+        logging.info("Generating UnitTest code coverage")
+
+        buildOutputBase = thebuilder.env.GetValue("BUILD_OUTPUT_BASE")
+        workspace = thebuilder.env.GetValue("WORKSPACE")
+
+        # Generate base code coverage for all source files
+        ret = RunCmd("lcov", f"--no-external --capture --initial --directory {buildOutputBase} --output-file {buildOutputBase}/cov-base.info --rc lcov_branch_coverage=1")

```

Comment:
Will {buildOutputBase} be resolved?

Reply:
Yes, it's working on my side.
![image](https://user-images.githubusercontent.com/96405988/200446572-1969b07a-77a6-4dbd-ad65-12f5e01bb213.png)